core/genesis.go

```diff
@@ -280,7 +280,13 @@ func SetupGenesisBlock(db ethdb.Database, triedb *triedb.Database, genesis *Gene
 	return SetupGenesisBlockWithOverride(db, triedb, genesis, nil)
 }
 
-func SetupGenesisBlockWithOverride(db ethdb.Database, triedb *triedb.Database, genesis *Genesis, overrides *ChainOverrides) (*params.ChainConfig, common.Hash, *params.ConfigCompatError, error) {
+func SetupGenesisBlockWithOverride(db ethdb.Database, triedb *triedb.Database, origGenesis *Genesis, overrides *ChainOverrides) (*params.ChainConfig, common.Hash, *params.ConfigCompatError, error) {
```

Comment:
I think my pref would be to retain the original signature with `genesis` and have return value in the function be `out` or something since the function sig will show up in your LSP and `origGenesis` might be confusing.

Should also be some sort of doc for this function which also mentions the original genesis is not modified?